libs/freelan/src/core.cpp

```diff
@@ -1552,6 +1545,15 @@ namespace freelan
 				m_logger(fscp::log_level::information) << "No IPv6 address configured.";
 			}
 
+			// If we are running in tun mode, we need at least one
+			if (m_configuration.tap_adapter.type == tap_adapter_configuration::tap_adapter_type::tun)
+			{
+				if (m_configuration.tap_adapter.ipv4_address_prefix_length.is_null() && m_configuration.tap_adapter.ipv6_address_prefix_length.is_null())
+				{
+					throw std::runtime_error("Running in tun mode, but no IPv4 or IPv6 address provided. Unable to continue without IPv4 and/or IPv6 address being configured.");
```

Comment:
I would go with:

    "Running in tun mode, but no IPv4 or IPv6 address was provided. Please configure at least one IPv4 or IPv6 address."

So that it doesn't only say what's wrong but also what to fix.